plugins/xrandr/msd-xrandr-manager.c

```diff
@@ -1671,73 +1671,44 @@ status_icon_popup_menu_selection_done_cb (GtkMenuShell *menu_shell, gpointer dat
         mate_rr_labeler_get_rgba_for_output (priv->labeler, output, &color);
 
-        cairo_set_source_rgb (cr, 0, 0, 0);
-        cairo_set_line_width (cr, OUTPUT_TITLE_ITEM_BORDER);
-        gtk_widget_get_allocation (widget, &allocation);
-        cairo_rectangle (cr,
-                         allocation.x + OUTPUT_TITLE_ITEM_BORDER / 2.0,
-                         allocation.y + OUTPUT_TITLE_ITEM_BORDER / 2.0,
-                         allocation.width - OUTPUT_TITLE_ITEM_BORDER,
-                         allocation.height - OUTPUT_TITLE_ITEM_BORDER);
-        cairo_stroke (cr);
-
-        gdk_cairo_set_source_rgba (cr, &color);
-        cairo_rectangle (cr,
-                         allocation.x + OUTPUT_TITLE_ITEM_BORDER,
-                         allocation.y + OUTPUT_TITLE_ITEM_BORDER,
-                         allocation.width - 2 * OUTPUT_TITLE_ITEM_BORDER,
-                         allocation.height - 2 * OUTPUT_TITLE_ITEM_BORDER);
-
-        cairo_fill (cr);
-
-        /* We want the label to always show up as if it were sensitive
-         * ("style->fg[GTK_STATE_NORMAL]"), even though the label is insensitive
-         * due to being inside an insensitive menu item.  So, here we have a
-         * HACK in which we frob the label's state directly.  GtkLabel's expose
-         * handler will be run after this function, so it will think that the
-         * label is in GTK_STATE_NORMAL.  We reset the label's state back to
-         * insensitive in output_title_label_after_expose_event_cb().
-         *
-         * Yay for fucking with GTK+'s internals.
-         */
+        color_string = gdk_rgba_to_string (&color);
 
-        gtk_widget_set_state (widget, GTK_STATE_NORMAL);
+        /*This can be overriden by themes, check all label:insensitive entries if it does not show up*/
+        string = g_string_new(NULL);
+        g_string_append (string, ".mate-panel-menu-bar menuitem.xrandr-applet:disabled>box>label{\n");
+        /* g_string_append (string, "color: black;"); Does not work-overridden in all themes*/
+        g_string_append (string, "padding-left: 4px; padding-right: 4px;");
+        g_string_append (string, "border-color: gray;");
+        g_string_append (string, "background-color:");
+        g_string_append (string, color_string);
+        g_string_append (string," }");
 
-        return FALSE;
-}
+        css = g_string_free (string, FALSE);
 
-/* See the comment in output_title_event_box_expose_event_cb() about this funny label widget */
-static gboolean
-output_title_label_after_draw_cb (GtkWidget *widget, cairo_t *cr, gpointer data)
-{
-        g_assert (GTK_IS_LABEL (widget));
-        gtk_widget_set_state (widget, GTK_STATE_INSENSITIVE);
+        context = gtk_widget_get_style_context (widget);
+        provider = gtk_css_provider_new ();
+        gtk_css_provider_load_from_data (provider,css, -1, NULL);
+
+	    gtk_style_context_add_provider (context,
```

Comment:
Indentation? Anyway I think it is not good idea to do this from draw callback. How often it is called? Don't remember, but changing/adding style could trigger new redraw...